tcadmin/resources/util.py

```diff
@@ -5,18 +5,14 @@
 # obtain one at http://mozilla.org/MPL/2.0/.
 
 from ..util.json import pretty_json
-
-# utilities for resource attr.ib's
-DESCRIPTION_PREFIX = """\
-*DO NOT EDIT* - This resource is configured automatically.
-
-"""
+from ..appconfig import AppConfig
 
 
 def description_converter(value):
     """Prepend *DO NOT EDIT* and a short explainer to the given value"""
+    description_prefix = AppConfig.current().description_prefix
     if not value.startswith("*DO NOT EDIT*"):
-        value = DESCRIPTION_PREFIX + value
+        value = description_prefix + value
```

Comment:
If `description_prefix` doesn't begin with `*DO NOT EDIT*`, then this will prepend that description prefix *again* on every run.  How could we avoid that?

Reply:
We can check it, if it starts with `*DO NOT EDIT*`.
Will have to check it

Reply:
I don't think we will have to check, because we are already checking the variable `value` before appending the prefix.
Am I correct?

Reply:
The transformation needs to be [idempotent](https://en.wikipedia.org/wiki/Idempotence), as it is applied to descriptions repeatedly.  This is because we call `description_converter` even when loading resources that already exist (and thus already have the description prefix in place).

If the function is not idempotent, then it will add the prefix *again* when loading such already-prefixed resources.

Reply:
I am thinking to add an `add` clause, with the condition 
```python
not description_prefix.startswith("*DO NOT EDIT*")
```

Reply:
Let's see what that would look like.

I wrote up some test cases for `description_converter` to ensure it's idempotent:

```
from tcadmin.resources.util import description_converter


def test_description_converter(appconfig):
    "Descriptions are prefixed"
    appconfig.description_prefix = 'I AM A PREFIX\n\n'
    assert description_converter('I am a description') == 'I AM A PREFIX\n\nI am a description'

def test_description_converter_idempotent(appconfig):
    "Descriptions are prefixed only once"
    appconfig.description_prefix = 'I AM A PREFIX\n\n'
    descr = 'I am a description'
    descr = description_converter(descr)
    descr = description_converter(descr)
    assert descr == 'I AM A PREFIX\n\nI am a description'
```
feel free to copy those into your PR.